axelrod/tests/test_tournament.py

```diff
@@ -29,28 +57,244 @@ def test_init(self):
+            processes=2)
+        tournament._run_parallel_repetitions(payoffs_list)
+        self.assertEqual(len(payoffs_list), self.test_repetitions)
+        for r in range(self.test_repetitions):
+            self.assertEqual(payoffs_list[r], self.expected_payoffs)
 
-        P1 = axelrod.Cooperator()
-        P2 = axelrod.TitForTat()
-        P3 = axelrod.Defector()
-        P4 = axelrod.Grudger()
-        P5 = axelrod.GoByMajority()
-        tournament = axelrod.Tournament(name='test', players=[P1, P2, P3, P4, P5], game=self.game, turns=200, repetitions=5)
-        names = [str(p) for p in tournament.players]
-        results = tournament.play().results
-        scores = [[sum([r[i] for ir,r in enumerate(res) if ir != ires]) for i in range(5)] for ires,res in enumerate(results)]
-        self.assertEqual(sorted(zip(names, scores)), outcome)
+    def test_n_workers(self):
+        max_processes = multiprocessing.cpu_count()
 
+        tournament = axelrod.Tournament(
+            name=self.test_name,
+            players=self.players,
+            game=self.game,
+            turns=200,
+            repetitions=self.test_repetitions,
+            processes=1)
+        self.assertEqual(tournament._n_workers(), max_processes)
 
-if __name__ == '__main__':
-    unittest.main()
+        tournament = axelrod.Tournament(
+            name=self.test_name,
+            players=self.players,
+            game=self.game,
+            turns=200,
+            repetitions=self.test_repetitions,
+            processes=max_processes + 2)
+        self.assertEqual(tournament._n_workers(), max_processes)
+
+    @unittest.skipIf(
+        multiprocessing.cpu_count() < 2,
+        "not supported on single processor machines")
+    def test_2_workers(self):
+        # This is a separate test with a skip condition because we
+        # cannot guarantee that the tests will always run on a machine
+        # with more than one processor
+        tournament = axelrod.Tournament(
+            name=self.test_name,
+            players=self.players,
+            game=self.game,
+            turns=200,
+            repetitions=self.test_repetitions,
+            processes=2)
+        self.assertEqual(tournament._n_workers(), 2)
+
+    def test_start_workers(self):
+        workers = 2
+        work_queue = multiprocessing.Queue()
+        done_queue = multiprocessing.Queue()
+        for repetition in range(self.test_repetitions):
+            work_queue.put(repetition)
+        tournament = axelrod.Tournament(
+            name=self.test_name,
+            players=self.players,
+            game=self.game,
+            turns=200,
+            repetitions=self.test_repetitions)
+        tournament._start_workers(workers, work_queue, done_queue)
+
+        stops = 0
+        while stops < workers:
+            payoffs = done_queue.get()
+            if payoffs == 'STOP':
+                stops += 1
+        self.assertEqual(stops, workers)
+
```

Comment:
`test_start_workers` above doesn't count the number of processes spawned by `tournament.start_workers` as that's well nigh impossible.

However, `tournament.start_workers` puts one `'STOP'` on to the work queue each time it spawns a process. This test checks that we have the expected number of STOPS and assumes that mulitprocessing correctly spawned the processes to go with them.